RevenueCatUI/CustomerCenter/ViewModels/ManageSubscriptionsViewModel.swift

```diff
@@ -84,13 +84,23 @@ class ManageSubscriptionsViewModel: ObservableObject {
 
     private func loadSubscriptionInformation() async throws {
         let customerInfo = try await purchasesProvider.customerInfo()
-        guard let currentEntitlementDict = customerInfo.entitlements.active.first,
-              let subscribedProductID = customerInfo.activeSubscriptions.first,
-              let subscribedProduct = await purchasesProvider.products([subscribedProductID]).first else {
+        
+        // Pick the soonest expiring iOS App Store entitlement and accompanying product.
+        guard let currentEntitlement = customerInfo.entitlements
+            .active
+            .values
+            .lazy
+            .filter({ entitlement in entitlement.store == .appStore })
+            .sorted(by: { lhs, rhs in
+                let lhsDateSeconds = lhs.expirationDate?.timeIntervalSince1970 ?? TimeInterval.greatestFiniteMagnitude
+                let rhsDateSeconds = rhs.expirationDate?.timeIntervalSince1970 ?? TimeInterval.greatestFiniteMagnitude
```

Comment:
Lifetime subscriptions are last in line, but this current implementation is not great because the order is undefined for multiple lifetime subscriptions. We can discuss options in the sync (see **Not in this PR** in the PR description),  or feel free to suggest any quick wins here.

Reply:
we usually go with the one that's expiring latest into the future (that's how we compute customerInfo for example), so that if you're subscribed to a monthly that's about to expire and have moved into a yearly or something, we show you the one that'll matter most - the one that gives you access for the longest time

Reply:
Yea that makes a lot of sense to me, but I was going off of this requirement on Linear:
>If there are more than once active subscription active, we pick the iOS subscription with the soonest non-null expiration date.

@vegaro Could you clarify/confirm which is the one we want?

Reply:
My thinking for this one was that let's say someone has one expiring tomorrow and another one further in the future, it is more relevant to display there's a subscription expiring soon. But from a length of access perspective, it's true the latest is more relevant. 

@aboedo when you say "that's how we compute customerInfo for example", what part of customer info you refer to? The expiration date of an entitlement?

I have been thinking and I think that what @aboedo is saying makes sense, since I want to see until when I have access to this (the one that gives you access for the longest time), but I also think that not displaying the soonest one (one that might charge the user tomorrow for example) might provide false information and cause more support issues than this view would solve. If I am a user and see that one of my subscriptions expires in a year, and I get charged tomorrow for another sub I would get mad and for sure contact support.

But it's true that showing soon-expiring subscriptions might trigger more cancellations.

The ideal solution is to display all entitlements I think, but for this first iteration, I think we have to pick what to display. What do you think about changing the perspective of this to be length of access (I think that's what we do in the dashboard) and instead of displaying "Expires on 12/30/2024" we display "Access until 12/30/2024" with the one with longest expiration?

Thoughts? I might be overthinking this...

Reply:
Let's go with that for now and get feedback on it. We can always change the logic. And yes, I think at some point 
 we should add another screen with the whole purchase history

Reply:
Alright, since that's a UI change I'll do that in a follow-up PR with more UI stuff.